### PR TITLE
feat: replace `pip` with `poetry`

### PR DESCRIPTION
**Pull request by martbern**
* Downloads and installs packages in parallel
* `poetry init` allows for meaningful package setup (e.g. automatically asks for package name etc.)
* `poetry install` both updates dependencies and installs current project as editable
* `poetry add [package-name]` automatically adds dependencies to the `pyproject.toml`
* Doesn't need a `setup.py`, all info is in `pyproject.toml`

You guys agree on migrating? :-)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+[tool.poetry.dependencies]
+python = ">=3.8,<3.11"
+black = "^22.3.0"
+numpy = "^1.22.4"
+scipy = "^1.8.1"
+pandas = "^1.4.2"
+SQLAlchemy = "^1.4.37"
+pyodbc = { version = "^4.0.32", markers = "platform_machine == 'x86_64'" }
+pre-commit = "^2.19.0"
+pytest = "^7.1.2"
```
Comment (KennethEnevoldsen) [inline pyproject.toml:16]: Is this everything installed when I install the package? I def. don't want that.

File: setup.cfg [hunk 2]
```diff
@@ -1,72 +0,0 @@
...
-    Intended Audience :: Developers
-    Intended Audience :: Science/Research
-    Topic :: Scientific/Engineering
-    Topic :: Scientific/Engineering :: Artificial Intelligence
-    Operating System :: POSIX :: Linux
-    Operating System :: MacOS :: MacOS X
-    Operating System :: Microsoft :: Windows
-    Programming Language :: Python :: 3.7
-    Programming Language :: Python :: 3.8
-    Programming Language :: Python :: 3.9
```
Comment (KennethEnevoldsen) [inline setup.cfg:17]: important metadata is not included.

File: setup.cfg [hunk 3]
```diff
@@ -1,72 +0,0 @@
...
-    # Don't complain about missing debug-only code:
-    def __unicode__
-    def __repr__
-    if self\.debug
-    # Don't complain if tests don't hit defensive assertion code:
-    raise AssertionError
-    raise NotImplementedError
-    # Don't complain if non-runnable code isn't run:
-    if 0:
-    if __name__ == .__main__.:
```
Comment (KennethEnevoldsen) [inline setup.cfg:70]: test coverage config is not included

**Review by KennethEnevoldsen — CHANGES_REQUESTED** [review]
I feel like switching to poetry could def. be worthwhile. However, we e.g. should never introduce a duplicate version number.

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+[tool.poetry]
+name = "psycopmlutils"
+version = "0.1.0"
```
[the remark this answers is not in the document]
Comment (martbern) [reply pyproject.toml:3]: Sure! We have nothing in prod, so not terribly important, but good to enforce now. I'll update the version. 

Was looking into github actions that might update this automatically based on e.g. "minor/major" in commit names. Or maybe this? https://github.com/mtkennerly/poetry-dynamic-versioning

Seems like something worth automating!

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+description = "A collection of machine-learnooing utilities used across the psycop-projects."
+authors = ["Martin Bernstorff <martinbernstorff@gmail.com>"]
```
[the remark this answers is not in the document]
Comment (martbern) [reply pyproject.toml:5]: Sure, whatever you guys want. So far I see this as internal tooling – adding more authors is absolutely no problem for me, though.

File: pyproject.toml (hunk 1 above)
Comment (martbern) [reply pyproject.toml:16]: Ah, yes! There's a dev subsection, will add it to the next commit.

File: setup.cfg (hunk 2 above)
Comment (martbern) [reply setup.cfg:17]: "Important" might be an overstatement for internal tooling 😉 

I've added it in the next commit 👍

File: setup.cfg [hunk 6]
```diff
@@ -1,72 +0,0 @@
...
-
-[flake8]
-ignore = E203, E266, E501, E731, W503
-max-line-length = 88
-max-complexity = 10
-select = B,C,E,F,W,T4,B9
-exclude =
-    .env,
-    .git,
-    __pycache__,
```
[the remark this answers is not in the document]
Comment (martbern) [reply setup.cfg:43]: Requires separate setup: https://dev.to/bowmanjd/using-flake8-and-pyproject-toml-with-flakehell-1cn1

Does the flake8 codestyle add much that we care about beyond black, though? :-)

File: .github/workflows/publish_to_pypi.yml [hunk 7]
```diff
@@ -1,34 +0,0 @@
-name: Publish to PyPI
```
[the remark this answers is not in the document]
Comment (martbern) [reply .github/workflows/publish_to_pypi.yml:1]: Don't think we want to publish to pypi just yet – but if we're starting to be disciplined re: versioning, then it's probably not a bad idea.

**Comment by HLasse** [comment]
We should probably add a small FAQ in the readme on how to use poetry (e.g. when to use the poetry commands instead of pip etc)

**Review by HLasse — APPROVED** [review]

**Review by KennethEnevoldsen — APPROVED** [review]

**Pull request merged**